dask/dataframe/tests/test_io.py

```diff
@@ -820,15 +832,88 @@ def test_to_csv():
         a = dd.from_pandas(df, npartitions)
         with tmpfile('csv') as fn:
             a.to_csv(fn)
-            result = pd.read_csv(fn, index_col=0)
+            result = dd.read_csv(fn).set_index('Unnamed: 0').compute()
```

Comment:
I'm not experiencing that difficulty, and that seems to be the status quo, but there are a lot of tests there, i'll split them to several functions.